shipyard_blueprints/microsoft-onedrive/shipyard_microsoft_onedrive/drive_client.py

```diff
@@ -246,139 +296,121 @@ def get_folder_id(self, folder_name: str, drive_id: Optional[str]) -> Optional[s
-            handle_errors(response.text, response.status_code)
+            raise e
+        else:
+            logger.info(f"Successfully created folder '{folder_name}' in OneDrive")
+            return response["id"]
 
-    def get_file_matches(
-        self, folder: str, pattern: str, drive_id: Optional[str]
-    ) -> List[str]:
-        """Returns a list of files that match the given pattern
+    def get_file_matches(self, folder: str, pattern: str, drive_id: str) -> List[str]:
+        """Returns a list of files that match the given pattern.
 
         Args:
-            pattern: The pattern to match on
-            drive_id: The ID of the drive to search in (only necessary if using basic authentication)
-
-        Returns: A list of files that match the pattern. If no matches are found then an empty list will be returned
+            folder (str): The folder to search in.
+            pattern (str): The pattern to match on.
+            drive_id (str): The ID of the drive to search in.
 
+        Returns:
+            List[str]: A list of files that match the pattern. If no matches are found, an empty list will be returned.
         """
-        url = f"{self.base_url}/drives/{drive_id}/root:/{folder}:/children"
+        matching_files = []
 
-        headers = {"Authorization": f"Bearer {self.access_token}"}
+        response = self._request("GET", f"drives/{drive_id}/root:/{folder}:/children")
 
-        matching_files = []
-        response = requests.get(url, headers=headers)
-        if response.ok:
-            logger.debug("Successfully fetched files from OneDrive")
-            files = response.json().get("value", [])
-            regex = re.compile(pattern)
-            for file in files:
-                if "name" in file and regex.search(file["name"]):
-                    matching_files.append(file)
+        logger.debug("Successfully fetched files from OneDrive")
+        files = response.get("value", [])
+        regex = re.compile(pattern)
+        matching_files.extend(
+            file for file in files if "name" in file and regex.search(file["name"])
+        )
 
-        else:
-            logger.warning(
-                f"Failed to fetch file matches from OneDrive: {response.text}"
-            )
         return matching_files
 
     def download_from_graph_url(self, download_url: str, file_name: str):
-        """Download a file from a given URL
+        """Download a file from a given URL.
 
         Args:
-            download_url: The download URL from the Graph API
-            file_name: The name of the file to write to
+            download_url (str): The download URL from the Graph API.
+            file_name (str): The name of the file to write to.
 
         Raises:
-            BadRequestError:
+            ExitCodeException: If the download fails.
         """
-        response = requests.get(download_url)
+        response = request("GET", download_url)
```

Comment:
I would have just left it as `requests.get`

Reply:
I changed how the module was imported so this was changed to reflect that